pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -507,12 +516,98 @@ func updateCertTimer(configInterval uint32, tickerHandle interface{}) {
 	flextimer.TickNow(tickerHandle)
 }
 
+func decryptAuthContainer(getconfigCtx *getconfigContext, sendRV *zedcloud.SendRetval) error {
+	decryptCtx := &cipher.DecryptCipherContext{
+		Log:                  log,
+		AgentName:            agentName,
+		PubSubControllerCert: getconfigCtx.pubControllerCert,
+		PubSubEdgeNodeCert:   getconfigCtx.zedagentCtx.subEdgeNodeCert,
+	}
+	sm := &zauth.AuthContainer{}
+	err := proto.Unmarshal(sendRV.RespContents, sm)
+	if err != nil {
+		return err
+	}
+	cipherBlock := sm.GetCipherData()
+	if cipherBlock == nil {
+		// Nothing encrypted, nothing to do
+		return nil
+	}
+	// Verify auth container and propagate the certificate
+	// errors to the caller by updating the retval status.
+	// That is utterly important that a caller schedules
+	// certs update in that case.
+	status, err := zedcloud.VerifyAuthContainer(zedcloudCtx, sm)
+	if err != nil {
+		sendRV.Status = status
+		return err
+	}
+	cipherContext := sm.GetCipherContext()
+	if decryptCtx == nil {
+		err := errors.New("decrypt cipher context is undefined\n")
+		return err
+	}
+	if cipherContext == nil {
+		err := errors.New("cipher context is undefined\n")
+		return err
+	}
+	if len(cipherBlock.CipherData) == 0 ||
+		len(cipherBlock.CipherContextId) == 0 {
+		err := errors.New("cipher block data or context id are incorrect\n")
+		return err
+	}
+	if cipherContext.ContextId != cipherBlock.CipherContextId {
+		err := errors.New("cipher context ids do not match\n")
+		return err
+	}
+	cipherCtx := &types.CipherContext{
+		ContextID:          cipherContext.GetContextId(),
+		HashScheme:         cipherContext.GetHashScheme(),
+		KeyExchangeScheme:  cipherContext.GetKeyExchangeScheme(),
+		EncryptionScheme:   cipherContext.GetEncryptionScheme(),
+		DeviceCertHash:     cipherContext.GetDeviceCertHash(),
+		ControllerCertHash: cipherContext.GetControllerCertHash(),
+	}
+	cipherBlockSt := types.CipherBlockStatus{
+		// No unique key is needed here, because this status is never published
+		CipherBlockID:   "cipher-block",
+		CipherContextID: cipherBlock.GetCipherContextId(),
+		InitialValue:    cipherBlock.GetInitialValue(),
+		CipherData:      cipherBlock.GetCipherData(),
+		ClearTextHash:   cipherBlock.GetClearTextSha256(),
+		CipherContext:   cipherCtx,
+		IsCipher:        true,
+	}
+	clearBytes, err := cipher.DecryptCipherBlock(decryptCtx, cipherBlockSt)
+	if err != nil {
+		return err
+	}
+
+	// Restore payload with decrypted bytes
+	sm.ProtectedPayload = &zauth.AuthBody{
+		Payload: clearBytes,
+	}
+	sm.CipherContext = nil
+	sm.CipherData = nil
+
+	bytes, err := proto.Marshal(sm)
```

Comment:
```suggestion
	smBytes, err := proto.Marshal(sm)
```

Reply:
Nope.

Reply:
will be renamed as soon as somebody wants to import `bytes`

Reply:
Agree with Christoph, names of standard packages should be avoided.